src/lib.rs

```diff
@@ -128,6 +131,32 @@ impl Default for State {
         }
     }
 }
+
+impl State {
+    /// Serializes the state.
+    pub fn serialize<W: io::Write>(&self, mut writer: W) -> io::Result<()> {
```

Comment:
Please use `where` clause consistently in this library.
`where` clause is more expressive than using normal syntax.